apps/condo/domains/ticket/access/TicketCommentFile.js

```diff
@@ -0,0 +1,110 @@
+// @ts-nocheck
+/**
+ * Generated by `createschema ticket.TicketCommentFile 'organization:Relationship:Organization:CASCADE;file?:File;ticketComment?:Relationship:TicketComment:SET_NULL'`
+ */
+
+const { throwAuthenticationError } = require('@condo/domains/common/utils/apolloErrorFormatter')
+const { RESIDENT, STAFF } = require('@condo/domains/user/constants/common')
+const { find, getByCondition, getById } = require('@core/keystone/schema')
+const compact = require('lodash/compact')
+const get = require('lodash/get')
+const { getTicketFieldsMatchesResidentFieldsQuery } = require('../utils/accessSchema')
+const { RESIDENT_COMMENT_TYPE, COMPLETED_STATUS_TYPE, CANCELED_STATUS_TYPE } = require('../constants')
+const uniq = require('lodash/uniq')
+const {
+    queryOrganizationEmployeeFor,
+    queryOrganizationEmployeeFromRelatedOrganizationFor, checkPermissionInUserOrganizationOrRelatedOrganization,
+} = require('@condo/domains/organization/utils/accessSchema')
+const omit = require('lodash/omit')
+const isEmpty = require('lodash/isEmpty')
+
+async function canReadTicketCommentFiles ({ authentication: { item: user } }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+
+    if (user.isSupport || user.isAdmin) return {}
+
+    if (user.type === RESIDENT) {
+        const residents = await find('Resident', { user: { id: user.id }, deletedAt: null })
+
+        const organizationsIds = compact(residents.map(resident => get(resident, 'organization')))
+        const residentAddressOrStatement = getTicketFieldsMatchesResidentFieldsQuery(user, residents)
+
+        return {
+            ticketComment: {
+                type: RESIDENT_COMMENT_TYPE,
+            },
+            ticket: {
+                organization: {
+                    id_in: uniq(organizationsIds),
+                    deletedAt: null,
+                },
+                OR: [
+                    { createdBy: { id: user.id } },
+                    ...residentAddressOrStatement,
+                ],
+            },
```

Comment:
can we use `client` here?

Reply:
yes, after https://github.com/open-condo-software/condo/pull/1381